src/main/java/com/hillel/items_exchange/service/UserService.java

```diff
@@ -101,20 +101,24 @@ private UserDto mapUserToDto(User user) {
         return convertToDto(parent.getChildren(), ChildDto.class);
     }
 
-    public void addChildren(User parent, List<ChildDto> childrenDtoToAdd) {
+    public List<ChildDto> addChildren(User parent, List<ChildDto> childrenDtoToAdd) {
         final List<Child> childrenToSave = new ArrayList<>(convertAllTo(
                 childrenDtoToAdd, Child.class, ArrayList::new));
         addNewChildren(parent, childrenToSave);
         userRepository.save(parent);
+        return convertToDto(findDuplicatedChildren(parent.getChildren(), childrenToSave), ChildDto.class);
     }
 
-    public void updateChildren(User parent, List<ChildDto> childrenDtoToUpdate) {
+    public List<ChildDto> updateChildren(User parent, List<ChildDto> childrenDtoToUpdate) {
         parent.getChildren().forEach(pChild -> childrenDtoToUpdate.forEach(uChild -> {
             if (pChild.getId() == uChild.getId()) {
                 BeanUtils.copyProperties(uChild, pChild);
             }
         }));
         userRepository.saveAndFlush(parent);
+        final List<Child> childrenToUpdate = new ArrayList<>(convertAllTo(
```

Comment:
The code above already performs an operation with `childrenDtoToUpdate` so why don't you just to all condition which would collect the same id and after get from user such updated children instead of making another convert operation here?

Reply:
I'm not sure I fully understand you, but I think my solution will work fine =)